test/spec/exerciseResult/exerciseResult.spec.js

```diff
@@ -353,4 +353,55 @@ describe('testing service "ExerciseResult":', function () {
             expect(exerciseResult.avgTimePerQuestion).toBe(expectedResult);
         });
     });
+
+    describe('test save exam isCompleted when all sections are completed',function(){
+
+        it('when getExerciseResult called with examSectionsNum as the forth param, then it should add examSectionsNum and examId on exerciseResult', function(){
+            var exerciseId = 4;
+            var examSectionsNum = 5;
+            var examId = 45;
+            var exerciseResult = actions.getExerciseResult(ExerciseTypeEnum.SECTION.enum, exerciseId, examId, examSectionsNum);
+            exerciseResult.$save();
+            $rootScope.$digest();
+            var examResult = actions.getExamResult(examId);
+            expect(exerciseResult.examId).toEqual(examId);
+            expect(examResult.examSectionsNum).toEqual(examSectionsNum);
+        });
+
+        it('when _saveExamWhenAllSectionsCompleted with exerciseNewStatus that is not 2, then getExamResult shouldn\'t been called inside the _saveExamWhenAllSectionsCompleted, total calls 1', function(){
```

Comment:
The required tests are: 
* Saving not completed section result -> exam is completed result should not be true.
* Saving completed section result while other exam sections are were not started-> exam is completed result should not be true.
* Saving completed section result while other exam sections are active but not completed -> exam is completed result should not be true.
* Saving completed section result while other exam sections are also completed -> exam is completed result should be true.